pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -498,20 +498,62 @@ func (c outboundListenerConflict) addMetric(push *model.PushContext) {
 // Connections to the ports of non-load balanced services are directed to
 // the connection's original destination. This avoids costly queries of instance
 // IPs and ports, but requires that ports of non-load balanced service be unique.
-func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env *model.Environment, node *model.Proxy, push *model.PushContext,
-	proxyInstances []*model.ServiceInstance, services []*model.Service) []*xdsapi.Listener {
+func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env *model.Environment, node *model.Proxy,
+	push *model.PushContext, proxyInstances []*model.ServiceInstance,
+	_ []*model.Service) []*xdsapi.Listener {
 
 	var proxyLabels model.LabelsCollection
 	for _, w := range proxyInstances {
 		proxyLabels = append(proxyLabels, w.Labels)
 	}
 
-	meshGateway := map[string]bool{model.IstioMeshGateway: true}
-	configs := push.VirtualServices(node, meshGateway)
+	sidecarScope := push.GetSidecarScope(node, proxyInstances)
+	services := sidecarScope.Services()
```

Comment:
Can this be removed since `push.SidecarScope will generate a default catch all egress listener.`?

Reply:
For this PR yes, but in the next PR or probably in the same PR, I need to handle this if block https://github.com/istio/istio/pull/10717/files#diff-86fb73d7dd150e47328ead76ceca0992R542

where user specifies list of listeners and services per listener. Wait for a day or two. I will complete this implementation and this code will become much more clear

Reply:
make sense